### PR TITLE
Fix incompatible features exception message

### PR DESCRIPTION
**Pull request by TayyabDev**
Addresses #10142 
Ex: server.xml with mphealth-1.0 mphealth-2.0
image

Ex: server.xml with jsf-2.2 cdi-1.0 cdi-2.0 mphealth-1.0 mphealth-2.0
image

### CODE REVIEW

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
Abandoning above build

**Comment by TayyabDev** [comment]
#build

File: dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops [hunk 1]
```diff
@@ -839,6 +839,10 @@ ERROR_CANT_RESOLVE_FEATURE_PROVIDE_LINK.useraction=Run featureUtility with a com
 
 MSG_USER_FEATURE_SERVER_XML=CWWKF1403I: The following user features are specified in the server.xml and they will not be installed: {0}
 
+ERROR_INCOMPATIBLE_FEATURES_SINGLETON=CWWKF1404E: The singleton features {0} and {1} are incompatible with each other and they cannot be installed at the same time.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.install/resources/com/ibm/ws/install/internal/resources/InstallKernel.nlsprops]: Change
"The singleton features {0} and {1} are incompatible with each other and they cannot be installed at the same time."
To
"**The {0} and {1} singleton features** are incompatible with each other and they cannot be installed at the same time."

**Comment by TayyabDev** [comment]
image
The fix works with the liberty maven plugin as well. Just need some more confirmation for the final exception message

**Comment by TayyabDev** [comment]
#build

**Comment by andrea-pichler** [comment]
Requested message review completed by L2

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java [hunk 2]
```diff
@@ -303,6 +347,61 @@ static InstallException create(RepositoryException e, Collection<String> feature
     static InstallException create(RepositoryResolutionException e, Collection<String> assetNames, File installDir, boolean installingAsset,
                                    boolean isOpenLiberty) {
         Collection<MissingRequirement> allRequirementsNotFound = e.getAllRequirementsResourcesNotFound();
+        if(allRequirementsNotFound.isEmpty()){
+            String msg = parseFeatureConflicts(e.getFeatureConflicts());
+            InstallException ie;
```
Comment (ericglau) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java]: Could make this one line 
```
InstallException ie = create(msg, e);
```

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java [hunk 3]
```diff
@@ -303,6 +347,61 @@ static InstallException create(RepositoryException e, Collection<String> feature
...
+                    StringBuilder nMinusOneFeatures = new StringBuilder();
+                    for(String feature : pair.subList(0, pair.size() - 1)){
+                        nMinusOneFeatures.append(feature).append(",");
+                    }
+                    String lastFeature = pair.get(pair.size() - 1);
+                    sb.append(Messages.INSTALL_KERNEL_MESSAGES.getMessage("ERROR_INCOMPATIBLE_FEATURES", nMinusOneFeatures.toString(), lastFeature)).append("\n");
+                }
+            }
+            InstallException ie;
+            ie = create(sb.toString(), e);
```
Comment (ericglau) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java]: Could make this one line 
```
InstallException ie = create(sb.toString(), e);
```

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java [hunk 4]
```diff
@@ -440,6 +542,13 @@ static InstallException create(RepositoryResolutionException e, Collection<Strin
     static InstallException create(RepositoryResolutionException e, Collection<String> assetNames, File installDir, boolean installingAsset,
                                    boolean isOpenLiberty, boolean isFeatureUtility) {
         Collection<MissingRequirement> allRequirementsNotFound = e.getAllRequirementsResourcesNotFound();
+        if(allRequirementsNotFound.isEmpty()){
+            String msg = parseFeatureConflicts(e.getFeatureConflicts());
+            InstallException ie;
+            ie = create(msg, e);
```
Comment (ericglau) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ExceptionUtils.java]: Could make this one line 
```
InstallException ie = create(msg, e);
```

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#run-libby-bot

**Comment by TayyabDev** [comment]
Build failed due to unrelated FATs failing. The install fat ran and passed.

**Pull request merged**